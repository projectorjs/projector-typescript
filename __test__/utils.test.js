const path = require("path");
const createProgramConfig = require("../utils").createProgramConfig;

describe("#createProgramConfig", () => {
  test("should load tsconfig and merge with provided compilerOptions and files", () => {
    return createProgramConfig({
      files: ["src/another.ts"],
      compilerOptions: {
        project: path.join(__dirname, "__mocks__", "load-config", "with-tsconfig"),
        sourceMap: true,
        declaration: true
      }
    }).then(config => expect(config).toMatchSnapshot());
  });

  test("should support tsconfig with extends", () => {
    return createProgramConfig({
      files: ["src/another.ts"],
      compilerOptions: {
        project: path.join(__dirname, "__mocks__", "load-config", "with-tsconfig-and-extends"),
        sourceMap: true
      }
    }).then(config => expect(config).toMatchSnapshot());
  });

  test("should not blow up if tsconfig doesn't exist", () => {
    const projectPath = path.join(__dirname, "__mocks__", "load-config", "without-tsconfig");
    return createProgramConfig({
      cwd: projectPath,
      files: [path.join(projectPath, "src/another.ts")],
      compilerOptions: {
        sourceMap: true,
        declaration: true
      }
    })
      .then(config => expect(config).toMatchSnapshot())
      .catch(error => {
        console.log(JSON.stringify(error, null, 2));
        throw error;
      });
  });

  test("should override compilerOptions from tsconfig with ones that were passed", () => {
    return createProgramConfig({
      cwd: path.join(__dirname, "__mocks__", "load-config", "with-tsconfig"),
      compilerOptions: {
        target: "es5"
      }
    }).then(config => expect(config).toMatchSnapshot());
  });
});

describe("#getAccessibleFileSystemEntries", () => {
  test();
});

describe("#splitDiagnosticsByType", () => {
  test();
});

describe("#groupDiagnosticsByFile", () => {
  test();
});

describe("#formatDiagnosticMessage", () => {
  test();
});
