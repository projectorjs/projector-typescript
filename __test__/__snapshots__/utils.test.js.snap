// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createProgramConfig should load tsconfig and merge with provided compilerOptions and files 1`] = `
Object {
  "compilerOptions": Object {
    "configFilePath": undefined,
    "declaration": true,
    "module": 5,
    "outDir": "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/dist",
    "project": "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig",
    "rootDir": "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/src",
    "sourceMap": true,
    "strict": true,
    "target": 2,
  },
  "files": Array [
    "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/src/another.ts",
  ],
}
`;

exports[`#createProgramConfig should not blow up if tsconfig doesn't exist 1`] = `
Object {
  "compilerOptions": Object {
    "configFilePath": undefined,
    "declaration": true,
    "sourceMap": true,
  },
  "files": Array [
    "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/without-tsconfig/src/another.ts",
  ],
}
`;

exports[`#createProgramConfig should override compilerOptions from tsconfig with ones that were passed 1`] = `
Object {
  "compilerOptions": Object {
    "configFilePath": undefined,
    "module": 5,
    "outDir": "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/dist",
    "rootDir": "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/src",
    "strict": true,
    "target": 1,
  },
  "files": Array [
    "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/src/index.ts",
    "/Users/ssysoev/Development/projector-typescript/__test__/__mocks__/load-config/with-tsconfig/src/utils.ts",
  ],
}
`;
